src/worker.cc

```diff
@@ -1120,6 +958,24 @@ std::string jsonify_error(const valhalla_exception_t& exception, const Api& requ
     body << (request.options().has_jsonp() ? request.options().jsonp() + "(" : "") << *json_error
          << (request.options().has_jsonp() ? ")" : "");
   }
+
+  // write a few stats about the error
+  request.mutable_info()->set_error(true);
+  auto worker = exception.code < 200 || (exception.code >= 300 && exception.code < 400)
+                    ? ".loki."
+                    : (exception.code >= 400 && exception.code <= 500 ? ".thor." : ".odin.");
+  const auto& action = Options_Action_Enum_Name(request.options().action());
+
+  auto* err_stat = request.mutable_info()->mutable_statistics()->Add();
+  err_stat->set_key(action + worker + exception.statsd_key);
```

Comment:
I don't know how generic this is, but at least Cloudwatch warns about creating high-cardinality metrics due to cost. It seems like this creates a different metric per exception which _may_ be costly in certain logging backends.

Reply:
it does however i expect the number of exceptional requests to remain relatively low. if that turns out not to be the case, what we can do is make use of the "frequency" option which only sends a percentage of the total metrics through to the backend. that should be doable pretty easily so i can just add it right now if you want!

Reply:
Not fully clear on how the frequency param would work. The cost I'm thinking of is per metric itself, and not per "publish" or "flush".

For example, if we end up pushing a total of 1000 different statsd keys, regardless of it's 1 or 100k samples pushed, that'd cost $300 per month in AWS as they charge $0.30 per metric and month

https://aws.amazon.com/cloudwatch/pricing/

I haven't counted the number of possible keys in this PR but it's something to be aware of I think. With that cost-calculation in mind, maybe something like a "blacklist" rather than a frequency param would be helpful, where the blacklist could be used for disabling certain groups of metrics that aren't of interest?

Reply:
ah yeah i didnt go check the docs on aws, i had figured it was by total number of metrics rather than by total number of unique keys. but yeah we could blacklist keys or we could prefix the exceptions with `error_`  and cut off everythign after the `error` part and collapse all exceptions into one key. it kind of neuters your ability to tell anything about what is going on though

Reply:
again i didnt read the aws docs (because they always give me headaches) but i think realistically only a very small percentage of these metrics will come out of a running service. you can grep your logs to see what they are. iirc its about 5 to 10 unique ones